### ISSUE TITLE
Code accepts mistyped named shelves

### ISSUE THREAD

**Issue by hwasiti** [andre-st/goodreads#11]
When copied the name of the shelf from GR and pasted in Bash, for some reason Bash will truncate the last letter and the code get the name wrong. However, this never happens when I write the shelf name manualy. So I suspect this is a problem with Ubuntu on Windows, but I don't remember this happened before.

Anyway, in the case of writing the wrong shelf name, the code will load ALL books. I think better, to respond with an error "Book shelf is not available", to prevent the misleading of loading the results of ALL books and the user thinks it is for a certain bookshelf.

By the way, the code is terrific. For all my books (~440 books), it took around 3-4 days, and without a single network problem. Resulted in 1018 GR users with at least 5% of the authors in 70429298's  shelf "%23ALL%23".

The cache helped a lot, when making other single shelf search or meta shelves (combination of 2 or more ) shelf search.

**Comment by andre-st** [comment]
Right, it's Goodreads' default behavior: You can reproduce this in your browser by selecting a shelf and then misspell the value of the "shelf" parameter in the URL. Goodreads won't show any error but the #ALL# shelf. 

I'll check how to warn about a misspelled shelf, It's a valid point.

**Comment by andre-st** [comment]
It's interesting to see your numbers. The cache is useful, but the dict-search slows down things. With every re-run it tries the next trigrams until it times out; you can deactivate it with --nodict.
